main.cpp

```diff
@@ -5173,18 +5173,63 @@ bool encrypt_command::execute(device_map &devices) {
         }
     }
 
+    uint8_t max_weight_delta = 1;
+
     if (!keyIsShare) {
         // Generate a random key share from 256-bit key
         std::random_device rand{};
         assert(rand.max() - rand.min() >= 256);
-        for(int i=0; i < 8; i++) {
-            for (int j=0; j < 12; j++) {
-                aes_key_share.bytes[i*16 + j] = rand();
-            }
-            aes_key_share.words[i*4 + 3] = aes_key.words[i]
-                                        ^ aes_key_share.words[i*4]
-                                        ^ aes_key_share.words[i*4 + 1]
-                                        ^ aes_key_share.words[i*4 + 2];
+        // Regenerate the shares until the hamming weights are close to each other
+        while (true) {
```

Comment:
I worry about whether this is guaranteed to terminate (may change depending on other comments)... how many loops does it do for you on average; I think we should limit and fail if it takes too many (or too long)